eclipse/src/saros/communication/SkypeManager.java

```diff
@@ -4,199 +4,48 @@
-      new XStreamExtensionProvider<String>(SarosPacketExtension.EXTENSION_NAMESPACE, "skypeInfo");
-
-  private final Map<JID, String> skypeNames = new ConcurrentHashMap<JID, String>();
-
-  private final XMPPConnectionService connectionService;
+  private final VersionManager versionManager;
   private final IPreferenceStore preferenceStore;
 
-  private PacketListener packetListener =
-      new PacketListener() {
-        @Override
-        public void processPacket(Packet packet) {
-
-          @SuppressWarnings("unchecked")
-          final XStreamIQPacket<String> iq = (XStreamIQPacket<String>) packet;
-
-          if (iq.getType() == IQ.Type.GET) {
-
-            final IQ reply = skypeProvider.createIQ(getLocalSkypeName());
-            reply.setType(IQ.Type.RESULT);
-            reply.setPacketID(iq.getPacketID());
-            reply.setTo(iq.getFrom());
-
-            final Connection connection = connectionService.getConnection();
-
-            if (connection == null) return;
-
-            try {
-              connection.sendPacket(reply);
-            } catch (IllegalStateException e) {
-              log.warn("failed to send IQ-RESULT reply to contact: " + iq.getFrom(), e);
-            }
-
-            return;
-          }
-
-          /* see hint for the class, maybe it is better to ignore result if there is already a name
-           * present and only force updates on SET
-           */
-          if (iq.getType() == IQ.Type.SET || iq.getType() == IQ.Type.RESULT) {
-
-            String skypeName = iq.getPayload();
-
-            if (skypeName != null && skypeName.trim().isEmpty()) skypeName = null;
-
-            if (skypeName != null) {
-              skypeNames.put(new JID(iq.getFrom()), skypeName);
-              log.debug("Skype Username for " + iq.getFrom() + " added: " + skypeName);
-            } else {
-              skypeNames.remove(new JID(iq.getFrom()));
-              log.debug("Skype Username for " + iq.getFrom() + " removed");
-            }
-          }
-        }
-      };
-
-  private final IRosterListener rosterListener =
-      new IRosterListener() {
-        @Override
-        public void presenceChanged(Presence presence) {
-
-          if (!presence.isAvailable()) return;
-
-          // TODO this also fires for AWAY, DND presence changes
-          final String from = presence.getFrom();
-
-          if (from == null) return;
-
-          requestSkypeName(new JID(from));
-        }
-      };
-
-  public SkypeManager(XMPPConnectionService connectionService, IPreferenceStore preferenceStore) {
-    this.connectionService = connectionService;
+  public SkypeManager(VersionManager versionManager, IPreferenceStore preferenceStore) {
+    this.versionManager = versionManager;
     this.preferenceStore = preferenceStore;
-    this.connectionService.addListener(this);
+
+    versionManager.setLocalInfo(PreferenceConstants.SKYPE_USERNAME, getLocalSkypeName());
```

Comment:
Seems we have to implement startable to the container. Actually this is "heavy" work, at least it seems so, because nothing it actually send at that point anyway. Maybe add an TODO ?

Reply:
Do you mean for the `preferenceStore.addPropertyChangeListener`?
We could add it but does it have a impact?